core/src/read/mod.rs

```diff
@@ -71,15 +76,17 @@ pub struct ReadStreamOptions<D: Decoder> {
 
     /// The number of prefetch blocks to store ahead of the cache block. This must be
     /// sufficiently large to ensure enough to time to fill the buffer in the worst
-    /// case latency scenerio.
+    /// case latency scenario.
     ///
     /// This should be left alone unless you know what you are doing.
+    ///
+    /// This will cause a panic if `num_cache_blocks + num_look_ahead_blocks < 3`.
     pub num_look_ahead_blocks: usize,
 
     /// The number of frames in a prefetch block.
     ///
     /// This should be left alone unless you know what you are doing.
-    pub block_size: usize,
+    pub block_frames: usize,
```

Comment:
This is a breaking change for the API and requires to bump the major version to 2.

Reply:
Hmm, if we're going to bump it to version 2, then I might as well implement some other API changes I had planned as well. In theory there's no reason this couldn't also be used for network streaming, so I want to make things more generic to accommodate that. I'll go ahead and work on that then.

Reply:
Though at first I think I'll at-least submit a PR that fixes the bug where the last remaining samples don't get written when a write stream is closed.

Reply:
From an outside perspective this is a good change, and better to do it the sooner rather than the later.